src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,129 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int ONE_ELEMENT = 1;
+    private static final double SIZE_GROWTH_INDEX = 1.5;
+
+    private Object[] elementData;
+    private int capacity = DEFAULT_CAPACITY;
+    private int size;
+
+    public ArrayList(int initialCapacity) {
+        if (initialCapacity >= 0 && initialCapacity < capacity) {
+            this.elementData = new Object[capacity];
+        } else if (initialCapacity > capacity) {
+            this.elementData = new Object[capacity];
+            this.elementData = grow();
+        } else {
+            throw new IllegalArgumentException("Illegal Capacity: "
+                    + initialCapacity);
+        }
+    }
+
+    public ArrayList() {
+        this.elementData = new Object[capacity];
+    }
+
     @Override
     public void add(T value) {
-
+        add(value,size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Incorrect index provided");
+        }
+        if (index > size && index < size + ONE_ELEMENT) {
+            size = index;
+        } else if (index > size + ONE_ELEMENT) {
+            throw new ArrayListIndexOutOfBoundsException("Incorrect index provided");
+        }
+        int newLength = size + ONE_ELEMENT;
+        if (newLength >= elementData.length || index > capacity) {
+            elementData = grow();
+        }
+        if (index != size + ONE_ELEMENT) {
+            System.arraycopy(elementData, index, elementData,
+                    index + ONE_ELEMENT, newLength - index);
+        }
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        validateIndex(index);
+        return (T) elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        validateIndex(index);
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        validateIndex(index);
+        final Object temp = elementData[index];
```

Comment:
`temp` is a bad name for variable. Rename it to `removedElement`

Reply:
done